code/addons/docs/src/plugins/mdx-plugin.ts

```diff
@@ -34,10 +33,7 @@ export async function mdxPlugin(options: Options): Promise<Plugin> {
       const mdxLoaderOptions = await options.presets.apply('mdxLoaderOptions', {
         ...mdxPluginOptions,
         mdxCompileOptions: {
-          providerImportSource: join(
-            dirname(require.resolve('@storybook/addon-docs/package.json')),
-            '/dist/shims/mdx-react-shim'
-          ),
+          providerImportSource: require.resolve('@storybook/mdx2-csf'),
```

Comment:
The reason for the funky join-with-dirname is to allow this to create an extensionless import.  If you simply `require.resolve` like this, the result will use the `require` condition from `@storybook/mdx2-csf`, and will point to the cjs entrypoint, even in the esm build.  So I think we need to keep using the more complicated strategy here.

Reply:
Since the plugin is now inside `@storybook/addons-docs`, should I just use the direct path to the package.json? 
Like this:
```ts
dirname(require.resolve('../../package.json')),'/dist/shims/mdx-react-shim'
```

Reply:
Yeah I think so, what do you say, @ndelangen?